src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,192 @@
+    {
+        int levelAtWhichToNotTriggerRaid = 1;
+        if (level == levelAtWhichToNotTriggerRaid)
+        {
+            return;
+        }
+
+        int numberOfBarbarians = level;
+        int numberOfArcherBarbarians = (int) (0.5 * level);
+        int numberOfChiefBarbarians = (int) (0.1 * level);
+
+        int hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+
+        if (hordeTotal > Configurations.maxBarbarianHordeSize)
+        {
+            final int maxSize = Configurations.maxBarbarianHordeSize;
+            if (hordeTotal > 40 && maxSize == 40)
+            {
+                numberOfBarbarians = 22;
+                numberOfArcherBarbarians = 16;
+                numberOfChiefBarbarians = 2;
+            }
+
+            numberOfBarbarians = hordeTotal - maxSize;
+
+            //For error handling and correct hordeTotal
+            if (numberOfBarbarians < 0)
+            {
+                numberOfBarbarians = 0;
+            }
+            hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+            if (hordeTotal > maxSize)
+            {
+                numberOfArcherBarbarians = hordeTotal - maxSize;
+
+                //For error handling and correct hordeTotal
+                if (numberOfArcherBarbarians < 0)
+                {
+                    numberOfArcherBarbarians = 0;
+                }
+                hordeTotal = numberOfArcherBarbarians + numberOfBarbarians + numberOfChiefBarbarians;
+                if (hordeTotal > maxSize)
+                {
+                    numberOfChiefBarbarians = hordeTotal - maxSize;
+
+                    //For error handling and correct hordeTotal
+                    if (numberOfChiefBarbarians < 0)
+                    {
+                        numberOfChiefBarbarians = 0;
+                    }
+                }
+            }
+        }
+
+        int x = colony.getCenter().getX();
+        int y = colony.getCenter().getY();
+        int z = colony.getCenter().getZ();
+
+        //Make sure world isn't null
+        if (raidingWorld == null)
+        {
+            return;
+        }
+        switch (raidingWorld.rand.nextInt(7))
+        {
+            case 0:
+                x += Configurations.workingRangeTownHall + 20;
+                break;
+            case 1:
+                x -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 2:
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 3:
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 4:
+                x += Configurations.workingRangeTownHall + 20;
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 5:
+                x += Configurations.workingRangeTownHall + 20;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)